src/lib/dapp/index.ts

```diff
@@ -208,14 +208,36 @@ export class DAppConnector {
   /**
    * Initiates the WalletConnect connection flow using a QR code.
    * @param pairingTopic - The pairing topic for the connection (optional).
+   * @param showErrorOnReject - Whether to show an error when the user rejects the pairing (default: false).
    * @returns {Promise<SessionTypes.Struct>} - A Promise that resolves when the connection process is complete.
    */
-  public async openModal(pairingTopic?: string): Promise<SessionTypes.Struct> {
+  public async openModal(
+    pairingTopic?: string,
+    showErrorOnReject: boolean = false,
```

Comment:
```suggestion
    throwErrorOnReject: boolean = false,
```
suggestion for better semantic clarity, as the exception handling is developer-dependent.